internal/versions/versions.go

```diff
@@ -338,6 +358,83 @@ func (h *handler) validateVersions() error {
 	return nil
 }
 
+func existAlready(db *gorm.DB, openShiftReleaseImage common.OpenShiftReleaseImage) bool {
```

Comment:
Why is this (both functions) part of this package/file? It seems independent to me.
Think about extracting this logic to a separate layer: this would improve testability (both of the extracted method and the current one) and maintanability.
If it will be necessary in versions package/method receiver we can always inject said indipendent layer

Reply:
I was not sure where is the best place to inject this code. I thought starting a new directory for this was a bit redundant, so I Just put it where I thought it is most fitting. Do you think I should create `internal/ocpversions` ?

Reply:
I think having small "classes" or equivalent in go is always good: it keeps the complexity low. If I have to pick, I'd rather have too many files than files with too many lines/responsibilities.
I think `internal/ocpversions` could be a good option, however the model is already named release_streams, WDYT about keeping it consistent with that?

Reply:
You mean release_images ?

Reply:
or release_source ?

Reply:
I meant release_sources , sorry